j7amo/shop-react-redux-cloudfront-be
# Task 5 integration with s3

[products_for_import.csv](https://github.com/j7amo/shop-react-redux-cloudfront-be/files/9225898/products_for_import.csv)
![Screenshot 2022-07-30 at 12 11 03](https://user-images.githubusercontent.com/66193591/181904447-05eddcf0-8fa9-4b2a-89bc-ba1e79342d8a.png)
**What's DONE:**

1 - File serverless.ts contains configuration for importProductsFile function
3 - The importProductsFile lambda function returns a correct response which can be used to upload a file into the S3 bucket
4 - Frontend application is integrated with importProductsFile lambda
5 - The importFileParser lambda function is implemented and serverless.ts contains configuration for the lambda

+1 (for JS only) - async/await is used in lambda functions

**NOT DONE:**

+1 (All languages) - importProductsFile lambda is covered by unit tests ((for JS only) aws-sdk-mock can be used to mock S3 methods - https://www.npmjs.com/package/aws-sdk-mock)
+1 (All languages) - At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder (move the file means that file should be copied into parsed folder, and then deleted from uploaded folder)

**HOW TO CHECK:**

To check **importProductsFile** function:

1) Go to Cloudfront distribution of frontend app and upload CSV file (you can find it in the attachment here): 

[https://d2g9tt19w7lruz.cloudfront.net/admin/products](https://d2g9tt19w7lruz.cloudfront.net/admin/products)

2) Check console logs via dev tools.

As for checking **importFileParser** it looks like you cannot do it outside of AWS console as it is triggered internally by S3 event. You can see the result of uploading a CSV file in the attachment screenshot.
